http-server/src/server.rs

```diff
@@ -86,6 +86,48 @@ impl Builder {
 
 	/// Register a new resource kind. Errors if `label` is already registered, or if number of
 	/// registered resources would exceed 8.
+	///
+	/// Every method will temporarily lock either the `default` or otherwise specified number of units during its
+	/// execution. Any method execution that would cause the total sum of locked resource units to exceed
+	/// the `capacity` of that resource will be denied execution, immediately returning JSON-RPC error object with code `-32604`.
+	///
+	/// ---
+	///
+	/// To specify a different than default number of units a method should use, use the `resources` argument in the
+	/// `#[method]` attribute:
+	///
+	/// ```
+	/// # use jsonrpsee::{types::RpcResult, proc_macros::rpc};
+	/// #
+	/// #[rpc(server)]
+	/// pub trait Rpc {
+	/// 	#[method(name = "my_expensive_method", resources("cpu" = 5, "mem" = 2))]
```

Comment:
> Either way; perhaps it'd be useful to make this clear?

I agree. It is not immediately obvious. I think one important piece to understand this is to first grok the way resources are set up: **not** here, around the trait marked with `#[rpc]` or at the method definition site; resources – along with their capacity and default cost – are set up on the **server** when it is constructed.

The reason for this separation is that the server app might want to make the resource set up configurable (through CLI or config file) so that different instances of the server can use different caps/defaults (e.g. fast servers == higher caps). It is also technically tricky (impossible?) to have a resource defined on the rpc modules and then somehow coalesced into a server-wide singleton.

So there's a UX problem here, but one we can't really fix. I think the docs should explain this important piece, as it makes understanding the rest much easier.